### ISSUE TITLE
[3.0.0][Bug] Jarhell when installing the plugin zip on opensearch core

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/security-analytics#1165]
https://build.ci.opensearch.org/blue/organizations/jenkins/distribution-build-opensearch/detail/distribution-build-opensearch/10068/pipeline/69
```
2024-07-16 18:05:24 INFO     Executing "/tmp/tmprdj4jneh/opensearch-3.0.0/bin/opensearch-plugin install --batch file:/tmp/tmprdj4jneh/opensearch-security-analytics-3.0.0.0.zip" in /tmp/tmprdj4jneh/opensearch-3.0.0

-> Installing file:/tmp/tmprdj4jneh/opensearch-security-analytics-3.0.0.0.zip

-> Downloading file:/tmp/tmprdj4jneh/opensearch-security-analytics-3.0.0.0.zip

-> Failed installing file:/tmp/tmprdj4jneh/opensearch-security-analytics-3.0.0.0.zip

-> Rolling back file:/tmp/tmprdj4jneh/opensearch-security-analytics-3.0.0.0.zip

-> Rolled back file:/tmp/tmprdj4jneh/opensearch-security-analytics-3.0.0.0.zip

Exception in thread "main" java.lang.IllegalStateException: failed to load plugin opensearch-security-analytics due to jar hell

	at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:693)

	at org.opensearch.plugins.InstallPluginCommand.jarHellCheck(InstallPluginCommand.java:861)

	at org.opensearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:829)

	at org.opensearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:874)

	at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:276)

	at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:250)

	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104)

	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)

	at org.opensearch.cli.MultiCommand.execute(MultiCommand.java:104)

	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)

	at org.opensearch.cli.Command.main(Command.java:101)

	at org.opensearch.plugins.PluginCli.main(PluginCli.java:66)

Caused by: java.lang.IllegalStateException: jar hell!

class: com.google.common.annotations.Beta

jar1: /tmp/tmprdj4jneh/opensearch-3.0.0/plugins/opensearch-job-scheduler/guava-32.1.3-jre.jar

jar2: /tmp/tmprdj4jneh/opensearch-3.0.0/plugins/opensearch-alerting/guava-32.0.1-jre.jar

	at org.opensearch.bootstrap.JarHell.checkClass(JarHell.java:316)

	at org.opensearch.bootstrap.JarHell.checkJarHell(JarHell.java:215)

	at org.opensearch.plugins.PluginsService.checkBundleJarHell(PluginsService.java:675)

	... 11 more

```

Thanks.

**Comment by peterzhuamazon** [comment]
Chain effect due to this one: 
* https://github.com/opensearch-project/alerting/issues/1599

**Comment by goyamegh** [comment]
Latest docker image publish worked for main: https://build.ci.opensearch.org/blue/organizations/jenkins/distribution-build-opensearch/detail/distribution-build-opensearch/10079/pipeline/167/

**Comment by goyamegh** [comment]
Started running security tests on a dummy PR to see if they are passing now: https://github.com/opensearch-project/security-analytics/actions/runs/9980393278?pr=1171

**Comment by goyamegh** [comment]
The security tests are failing due to assertion failures now for three tests specifically. I'm looking into them now, to figure out the root cause.

**Comment by goyamegh** [comment]
@peterzhuamazon jar hell issue is now fixed, which was verified by confirming that the docker image is getting published as expected. For integ test failures with security enabled, it is being tracked and fixed as part of: https://github.com/opensearch-project/security-analytics/issues/1182.

Let me know if I can close this issue now. Thanks!

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon jar hell issue is now fixed, which was verified by confirming that the docker image is getting published as expected. For integ test failures with security enabled, it is being tracked and fixed as part of: #1182.
> 
> Let me know if I can close this issue now. Thanks!

Go ahead and close the issue.
Thanks @goyamegh !
